pkg/cmd/workflow/run/run.go

```diff
@@ -217,7 +217,18 @@ func collectInputs(p iprompter, yamlContent []byte) (map[string]string, error) {
 
 	for _, input := range inputs {
 		var answer string
-		if input.Required {
+
+		if input.Type == "choice" {
+			description := input.Description
+			if input.Required {
+				description += " (required)"
+			}
+			selected, err := p.Select(description, "", input.Options)
+			if err != nil {
+				break
```

Comment:
This isn't correct. What this is saying is "if there is an error prompting, then `break` out of this loop" where breaking out of the loop results in the submission of the `providedInputs`. This is different than the `break` below which only breaks out of the `required` `for {}`. The error return actually happens on line 243.

I don't really like this pattern because it causes easy confusion as we've seen here. I think it's almost always better to return an error immediately. Please change this to:

```go
if err != nil {
   return nil, err
}
```

Reply:
An error can occur for example, if I hit `ctrl+C` while being prompted. I would expect that to exit the CLI but in this case it actually ends up submitting the workflow!